cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -169,26 +182,32 @@ MapBuilderBridge::GetTrajectoryStates() {
     const int trajectory_id = entry.first;
     const SensorBridge& sensor_bridge = *entry.second;
 
-    const cartographer::mapping::TrajectoryBuilder* const trajectory_builder =
-        map_builder_.GetTrajectoryBuilder(trajectory_id);
-    if (trajectory_builder == nullptr) {
-      continue;
+    std::unique_ptr<TrajectoryState::LocalSlamData>
+        current_trajectory_state_data;
+    {
+      cartographer::common::MutexLocker lock(&mutex_);
+      if(trajectory_state_data_.count(trajectory_id) == 0) {
+        continue;
+      }
+      current_trajectory_state_data =
+          cartographer::common::make_unique<TrajectoryState::LocalSlamData>(
+              trajectory_state_data_.at(trajectory_id));
     }
-    const cartographer::mapping::TrajectoryBuilder::PoseEstimate pose_estimate =
-        trajectory_builder->pose_estimate();
-    if (cartographer::common::ToUniversal(pose_estimate.time) < 0) {
+
+    const ::cartographer::common::Time& time =
+        current_trajectory_state_data->time;
+    if (cartographer::common::ToUniversal(time) < 0) {
```

Comment:
In which case would this happen?

Reply:
No idea. I preserver the existing line without thinking too much. Here's where it was added for the first time. You approved the PR :-)

https://github.com/googlecartographer/cartographer_ros/pull/53/files#diff-039590ba736b783b50a4b1f31851b06bR526

Reply:
I meant in which case does it happen with your change? Before there was of course the situation where a trajectory builder has been added but has not produced any output yet.